lib/study.js

```diff
@@ -0,0 +1,54 @@
+const self = require('sdk/self');
+const shield = require('shield-studies-addon-utils');
+const { when: unload } = require('sdk/system/unload');
+
+const { Advisor } = require('./Advisor');
+
+let Advise = null;
+
+const studyConfig = {
+  name: self.addonId,
+  duration: 7,
+  surveyUrls: {
+    'end-of-study': 'mozilla.org',
+    'user-ended-study': 'mozilla.org',
+    ineligible: null,
+  },
+  variations: {
+    'passive-panel'() {
+      Advise.start('passivePanel');
+    },
+    'auto-open-panel'() {
+      Advise.start('autoOpenPanel');
+    },
+    'observe-only'() {},
+  },
+};
+
+class SecurityAdvisorStudy extends shield.Study {
+  constructor(config) {
+    super(config);
+    Advise = new Advisor();
+  }
+  isEligible() {
+    return super.isEligible() && Advise.isEligible();
+  }
+  cleanup() {
+    if (this.variation !== 'observe-only') {
```

Comment:
In my study add-on, I didn't wrap the `cleanup` logic in an if-check. Is there a reason you did so here? The `Advise.cleanup()` method looks "safe" to run even for the `observe-only` users.